tests/integration-tests/tests/schedulers/test_slurm.py

```diff
@@ -1605,24 +1604,182 @@ def _test_memory_based_scheduling_enabled_false(
     assert_that(slurm_commands.get_node_attribute("queue1-st-ondemand1-i1-1", "Memory")).is_equal_to("3891")
     assert_that(slurm_commands.get_node_attribute("queue1-dy-ondemand1-i3-1", "Memory")).is_equal_to("31129")
 
-    # TODO: Add functional tests for memory-based scheduling
+    # Upload files for memory allocation tests
+    remote_command_executor._copy_additional_files(
+        [
+            str(test_datadir / "memory_allocation_chars.c"),
+        ],
+    )
+
+    # Compile C program to test memory allocations
+    remote_command_executor.run_remote_command("gcc memory_allocation_chars.c")
+    assert_that(remote_command_executor.run_remote_command("ls ./a.out").stdout).contains("a.out")
+
+    # Check that I can use the `--mem` flag to filter compute nodes
+    # Try to allocate on nodes with not enough memory
+    result = slurm_commands.submit_command(
+        nodes=1,
+        command="sleep 1",
+        constraint="ondemand1-i1",
+        other_options="--mem=4000 --test-only",
```

Comment:
Why should we include `--test-only`?
Is it like a dryrun?

Reply:
Yes. This way you can test the submission command without running the actual job. Here we are interested only in checking that the `--mem` flag works as intended, so it's faster to use `--test-only`.